sdk/src/trace/default_tracer_provider.cc

```diff
@@ -6,8 +6,8 @@ namespace sdk
 {
 namespace trace
 {
-DefaultTracerProvider::DefaultTracerProvider() noexcept
-    : tracer_(new Tracer(std::unique_ptr<Recorder>(nullptr)))
+DefaultTracerProvider::DefaultTracerProvider(std::unique_ptr<SpanProcessor> &&processor) noexcept
```

Comment:
Why would we want `unique_ptr` versus `shared_ptr` here?

Reply:
Changed to a `shared_ptr`.